File: learnmodel/src/main/java/com/sandgrains/edu/learnmodel/HtmlTextView.java
package com.sandgrains.edu.learnmodel;

import android.content.Context;
import android.text.Html;
import android.text.method.LinkMovementMethod;
import android.util.AttributeSet;
import android.widget.TextView;

import java.io.InputStream;

public class HtmlTextView extends androidx.appcompat.widget.AppCompatTextView {
    public static final String TAG = "HtmlTextView";
    public static final boolean DEBUG = false;

    public HtmlTextView(Context context, AttributeSet attrs, int defStyle) {
        super(context, attrs, defStyle);
    }

    public HtmlTextView(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    public HtmlTextView(Context context) {
        super(context);
    }

    /**
     * @param is
     * @return
     */
    static private String convertStreamToString(InputStream is) {
        java.util.Scanner s = new java.util.Scanner(is).useDelimiter("\\A");

        return s.hasNext() ? s.next() : "";
    }

    /**
     * Parses String containing HTML to Android's Spannable format and displays
     * it in this TextView.
     *
     * @param html String containing HTML, for example: "<b>Hello world!</b>"
     */
    public void setHtmlFromString(String html) {
        Html.ImageGetter imgGetter;

        imgGetter = new UrlImageGetter(this, getContext());


        // this uses Android's Html class for basic parsing, and HtmlTagHandler
        setText(Html.fromHtml(html, imgGetter, null));

        // make links work
        setMovementMethod(LinkMovementMethod.getInstance());
        setTextColor(getResources().getColor(android.R.color.secondary_text_dark_nodisable,null));
    }
}

File: student/src/main/java/com/sandgrains/edu/student/utils/custom/HtmlTextView.kt
package com.sandgrains.edu.student.utils.custom


import android.content.Context
import android.text.Html
import android.text.Html.ImageGetter
import android.text.method.LinkMovementMethod
import android.util.AttributeSet
import androidx.appcompat.widget.AppCompatTextView
import com.sandgrains.edu.student.ui.video.UrlImageGetter

class HtmlTextView @JvmOverloads constructor(
    context: Context,
    attrs: AttributeSet? = null,
    defStyleAttr: Int = 0
) : AppCompatTextView(context, attrs, defStyleAttr) {

    /**
     * Parses String containing HTML to Android's Spannable format and displays
     * it in this TextView.
     *
     * @param html String containing HTML, for example: "**Hello world!**"
     */
    fun setHtmlFromString(html: String?) {
        val imgGetter: ImageGetter
        imgGetter = UrlImageGetter(this, context)


        // this uses Android's Html class for basic parsing, and HtmlTagHandler
        text = Html.fromHtml(html, imgGetter, null)

        // make links work
        movementMethod = LinkMovementMethod.getInstance()
//        setTextColor(resources.getColor(R.color.secondary_text_dark_nodisable, null))
    }
}